src/en/manhwaread/src/eu/kanade/tachiyomi/extension/en/manhwaread/ManhwaRead.kt

```diff
@@ -0,0 +1,305 @@
+        val res = client.newCall(GET(url, headers)).execute()
+        val doc = Jsoup.parse(res.body.string(), baseUrl)
+
+        val anchors = doc.select("#chaptersList a.chapter-item").ifEmpty { doc.select("a[href*=/chapter-]") }
+        if (anchors.isEmpty()) {
+            return Observable.just(
+                listOf(
+                    SChapter.create().apply {
+                        name = "Chapter"
+                        this.url = manga.url
+                    },
+                ),
+            )
+        }
+
+        val chapters = anchors.asReversed().map { a ->
+            val name = a.selectFirst(".chapter-item__name")?.text()?.ifBlank { null } ?: a.text()
+            val href = a.attr("href")
+            val absHref = if (href.startsWith("http")) href else baseUrl + href
+            val dateText = a.selectFirst(".chapter-item__date")?.text()?.ifBlank { null }
+                ?: a.selectFirst("time[datetime]")?.attr("datetime")
+                ?: a.selectFirst(".chapter-release-date, span.chapter-release-date")?.text()
+            SChapter.create().apply {
+                this.name = name
+                this.url = absHref
+                this.date_upload = parseChapterDate(dateText)
+            }
+        }
+
+        return Observable.just(chapters)
+    }
+
+    override fun pageListRequest(chapter: SChapter): Request {
+        val url = if (chapter.url.startsWith("http")) chapter.url else baseUrl + chapter.url
+        return GET(url, headers)
+    }
+
+    override fun imageFromElement(element: Element): String? {
+        return when {
+            element.hasAttr("data-src") -> element.attr("abs:data-src")
+            element.hasAttr("data-lazy-src") -> element.attr("abs:data-lazy-src")
+            element.hasAttr("srcset") -> element.attr("abs:srcset").substringBefore(" ").removeSuffix(",")
+            element.hasAttr("data-cfsrc") -> element.attr("abs:data-cfsrc")
+            else -> element.attr("abs:src")
+        }
+    }
+
+    private inline fun <reified T> Response.parseAs(): T {
+        return json.decodeFromString(body.string())
+    }
+
+    public override fun parseChapterDate(raw: String?): Long {
+        if (raw.isNullOrBlank()) return 0L
+        val text = raw.trim()
+
+        val formats = listOf(
+            java.text.SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssXXX", java.util.Locale.US),
+            java.text.SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'", java.util.Locale.US),
+            java.text.SimpleDateFormat("yyyy-MM-dd", java.util.Locale.US),
+            java.text.SimpleDateFormat("dd/MM/yyyy", java.util.Locale.US),
+            java.text.SimpleDateFormat("MM/dd/yyyy", java.util.Locale.US),
+            java.text.SimpleDateFormat("yyyy/MM/dd", java.util.Locale.US),
+        )
+        for (fmt in formats) {
+            fmt.isLenient = false
+            kotlin.runCatching { return fmt.parse(text)?.time ?: 0L }
```

Comment:
```suggestion
            fmt.tryParse(text)
```
import from `keiyoushi.utils`

Reply:
removed